src/Proto.Cluster/PidCache.cs

```diff
@@ -25,6 +25,12 @@ public bool TryGet(ClusterIdentity clusterIdentity, out PID pid)
         {
             if (clusterIdentity is null) throw new ArgumentNullException(nameof(clusterIdentity));
 
+            if (clusterIdentity.CachedPid is { } identityCachedPid)
+            {
+                //If the PID is already cached using ClusterIdentity, we can skip the lookup altogether
+                pid = identityCachedPid;
```

Comment:
is this invalidated somewhere? so we don't get get stuck with an invalid pid at some point